pkg/interactor/list_data.go

```diff
@@ -10,94 +10,55 @@ import (
 
 // List provides all the thing's data owned by an user
 func (d *DataInteractor) List(token string, query *entities.Query) ([]entities.Data, error) {
-	data, err := d.getDevicesData(token, query)
+	things, err := d.things.List(token)
 	if err != nil {
-		return []entities.Data{}, err
+		return nil, fmt.Errorf("error getting list of things: %w", err)
 	}
 
-	data, err = d.filterDataBySensorID(data, query.ThingID, query.SensorID)
-	if err != nil {
-		return []entities.Data{}, err
+	if query.ThingID != "" {
+		things = []*btEntities.Thing{{ID: query.ThingID}}
 	}
 
-	return data, err
-}
-
-func (d *DataInteractor) getDevicesData(token string, query *entities.Query) ([]entities.Data, error) {
-	data := []entities.Data{}
-	things, err := d.things.List(token)
+	data, err := d.getAllData(things, query)
 	if err != nil {
-		return data, fmt.Errorf("%v: %w", ErrValidToken, err)
+		return nil, fmt.Errorf("error getting the data: %w", err)
 	}
 
-	if query.ThingID != "" {
-		err := d.verifyAuthorization(token, query.ThingID)
+	if query.SensorID != "" {
+		data, err = d.filterDataBySensorID(data, query.SensorID)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error filtering data by sensor ID: %w", err)
 		}
 	}
 
-	data, err = d.getAllData(things, query)
-	if err != nil {
-		return data, err
-	}
-
 	return data, nil
 }
 
 func (d *DataInteractor) getAllData(things []*btEntities.Thing, query *entities.Query) ([]entities.Data, error) {
 	data := []entities.Data{}
 	for _, t := range things {
 		query.ThingID = t.ID
-		fmt.Println(t.ID)
 		thingData, err := d.DataStore.Get(query)
 		if err != nil {
-			return data, fmt.Errorf("error getting data: %w", err)
+			return nil, fmt.Errorf("error getting data: %w", err)
 		}
-
 		data = append(data, thingData...)
 	}
-
 	return data, nil
 }
 
-func (d *DataInteractor) verifyAuthorization(token, id string) error {
```

Comment:
Why did you remove this operation? List things by using the passed token isn't enough because we need to verify if the `query.ThingID` matches with a thing owned by that user.

Reply:
Yeah, you're right. I've wrongly assumed that `query.ThingID` will always be related to the user token owner. Added this verifation, done!